src/components/Geolocation.tsx

```diff
@@ -0,0 +1,124 @@
+import { Matcher, SelectableStaticFilter, useSearchActions, useSearchState } from '@yext/search-headless-react';
+import { executeSearch } from '../utils/search-operations';
+import { getUserLocation } from '../utils/location-operations';
+import { useComposedCssClasses } from '../hooks/useComposedCssClasses';
+import { useCallback, useState } from 'react';
+import LoadingIndicator from '../icons/LoadingIndicator';
+import { YextIcon } from '../icons/YextIcon';
+
+/**
+ * The CSS class interface for the Geolocation component.
+ *
+ * @public
+ */
+export interface GeolocationCssClasses {
+  geolocationContainer?: string,
+  button?: string,
+  iconContainer?: string
+}
+
+const builtInCssClasses: Readonly<GeolocationCssClasses> = {
+  geolocationContainer: 'text-sm text-neutral text-center justify-center items-center flex flex-row',
+  button: 'text-primary font-semibold hover:underline focus:underline',
+  iconContainer: 'w-4 ml-2'
+};
+
+/**
+ * The props for the Geolocation component.
+ *
+ * @public
+ */
+export interface GeolocationProps {
+  /**
+   * Configuration used when collecting the user's location.
+   * Definition: {@link https://w3c.github.io/geolocation-api/#position_options_interface}.
+   */
+  geolocationOptions?: PositionOptions,
+  /**
+   * The radius, in miles, around the user's location to find results. Defaults to 50.
+   * If location accuracy is low, a larger radius may be used automatically.
+   */
+  radius?: number,
+  /** The label for the button. Defaults to 'Use my location'. */
+  label?: string,
+  /** Custom icon component to display along with the button. */
+  GeolocationIcon?: React.FC,
```

Comment:
not 100% sure but I think React.FunctionalComponent is preferred over React.FC

Reply:
sure! I can update

Reply:
Ah ok either way is fine then, I think I read some variant of this blog post a while back https://spin.atomicobject.com/2022/01/04/think-twice-react-fc/ but there's a lot of hot takes on the internet

Reply:
Reading it again I think the React.FC they're talking about (from the create-react-app template) is different than this one